docs/guides/tfhers.md

```diff
@@ -0,0 +1,242 @@
+
+2. The Concrete Optimizer doesn't limit its search space to available TFHE-rs parameters. Therefore, we need to either fix or limit the search space in areas of the circuit where compatibility is required. This will ensure that the same parameters are used where needed.
+
+## First Use Case: Shared Secret Key
+
+The first scenario involves a single party seeking to combine both Concrete and TFHE-rs in a computation. In this case, a shared secret key will be used, while different keysets will be held for Concrete and TFHE-rs. It is possible to generate keys in both directions, meaning that you can start by generating keys from Concrete and then use Concrete's secret key to generate the remaining keys (partial keygen) in TFHE-rs (use case 1.1), or vice versa (use case 1.2).
+
+After the key generation is complete and both keysets are acquired, we can perform computations, encryption, and decryption on both ends.
+
+![alt text](../_static/tfhers/tfhers_uc1.svg)
+
+### Initial Setup and Config
+
+The first step is to define the TFHE-rs ciphertext we want to use (see [The Basics](#the-basics)). This includes specifying cryptographic and encoding parameters.
+
+```python
+from functools import partial
+from concrete.fhe import tfhers
+
+tfhers_params = tfhers.CryptoParams(
+    lwe_dimension=909,
+    glwe_dimension=1,
+    polynomial_size=4096,
+    pbs_base_log=15,
+    pbs_level=2,
+    lwe_noise_distribution=0,  # this won't be needed so we set it to 0
+    glwe_noise_distribution=2.168404344971009e-19,
+    encryption_key_choice=tfhers.EncryptionKeyChoice.BIG,
+)
+# creating a TFHE-rs ciphertext type with crypto and encoding params
+tfhers_type = tfhers.TFHERSIntegerType(
+    is_signed=False,
+    bit_width=8,
+    carry_width=3,
+    msg_width=2,
+    params=tfhers_params,
+)
+# this partial will help us create TFHERSInteger with the given type instead of calling
+# tfhers.TFHERSInteger(tfhers_type, value) every time
+tfhers_int = partial(tfhers.TFHERSInteger, tfhers_type)
+```
+
+### Defining the Circuit and Compiling
+
+We will now define a simple addition function. This function takes TFHE-rs inputs, converts them to native (Concrete), runs a computation, and then converts them back to TFHE-rs. The circuit below is a common example that takes and produces TFHE-rs ciphertexts. However, there are other scenarios where you might not convert back to TFHE-rs, or you might convert to a different type than the input. Another possibility is to take one native ciphertext and one TFHE-rs ciphertext.
+
+```python
+
+def compute(x, y):
+    ####### TFHE-rs to Concrete #########
+    x = tfhers.to_native(x)  # the tfhers type will be derived from the x and y values
+    y = tfhers.to_native(y)
+    ####### TFHE-rs to Concrete #########
```

Comment:
if input x and y are "real integers" and then "x = tfhers.to_native(x)" is in TFHE-rs format, I would have different names for clarity. 

something like
x_concrete = tfhers.to_native(x_int)